pkg/monitor/monitor/template.go

```diff
@@ -154,23 +48,239 @@ var dashBoardConfig = `{
+	var c = config.Config{
+		AlertingConfig: config.AlertingConfig{
+			AlertRelabelConfigs: nil,
+			AlertmanagerConfigs: nil,
+			XXX:                 nil,
+		},
+		GlobalConfig: config.GlobalConfig{
+			ScrapeInterval:     model.Duration(15 * time.Second),
+			EvaluationInterval: model.Duration(15 * time.Second),
+		},
+		RuleFiles: []string{
+			"/prometheus-rules/rules/*.rules.yml",
+		},
+		ScrapeConfigs: []*config.ScrapeConfig{
+			{
+				JobName:        "tidb-cluster",
+				ScrapeInterval: model.Duration(15 * time.Second),
+				HonorLabels:    true,
+				ServiceDiscoveryConfig: config.ServiceDiscoveryConfig{
+					KubernetesSDConfigs: []*config.KubernetesSDConfig{
+						{
+							Role: "pod",
+							NamespaceDiscovery: config.KubernetesNamespaceDiscovery{
+								Names: cmodel.ReleaseNamespaces,
+							},
+						},
+					},
+				},
+				HTTPClientConfig: config.HTTPClientConfig{
+					TLSConfig: config.TLSConfig{
+						InsecureSkipVerify: true,
+					},
+					XXX: map[string]interface{}{
+						"scheme": "http",
+					},
+				},
+				RelabelConfigs: []*config.RelabelConfig{
+					{
+						SourceLabels: model.LabelNames{
+							instanceLabel,
+						},
+						Action: config.RelabelKeep,
+						Regex:  *cmodel.ReleaseTargetRegex,
+					},
+					{
+						SourceLabels: model.LabelNames{
+							scrapeLabel,
+						},
+						Action: config.RelabelKeep,
+						Regex:  truePattern,
+					},
+					{
+						SourceLabels: model.LabelNames{
+							metricsPathLabel,
+						},
+						Action:      config.RelabelReplace,
+						TargetLabel: "__metrics_path__",
+						Regex:       allMatchPattern,
+					},
+					{
+						SourceLabels: model.LabelNames{
+							namespaceLabel,
+						},
+						Action:      config.RelabelReplace,
+						TargetLabel: "kubernetes_namespace",
+					},
+					{
+						SourceLabels: model.LabelNames{
+							podNameLabel,
+						},
+						Action:      config.RelabelReplace,
+						TargetLabel: "instance",
+					},
+					{
+						SourceLabels: model.LabelNames{
+							instanceLabel,
+						},
+						Action:      config.RelabelReplace,
+						TargetLabel: "cluster",
+					},
+				},
+			},
+		},
+	}
+	return &c
+}
+
+func addAlertManagerUrl(pc *config.Config, cmodel *MonitorConfigModel) {
+	pc.AlertingConfig = config.AlertingConfig{
+		AlertmanagerConfigs: []*config.AlertmanagerConfig{
+			{
+				ServiceDiscoveryConfig: config.ServiceDiscoveryConfig{
+					StaticConfigs: []*config.TargetGroup{
+						{
+							Targets: []model.LabelSet{
+								map[model.LabelName]model.LabelValue{
+									"targets": model.LabelValue(cmodel.AlertmanagerURL),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
 }
 
-func renderTemplateFunc(tpl *template.Template, model interface{}) (string, error) {
-	buff := new(bytes.Buffer)
-	err := tpl.Execute(buff, model)
+func addTlsConfig(pc *config.Config, cmodel *MonitorConfigModel) {
+
+	r, _ := config.NewRegexp(".*\\-tikv\\-\\d*$")
```

Comment:
Do not ignore error, for compiling regexp from string literal, I recommend creating a package level variable in init block, so it can fail fast on start if the regex is wrong

Reply:
updated.

Reply:
I didn't see how it was updated